ocaml/xapi/xapi_vm.ml

```diff
@@ -238,6 +238,32 @@ let assert_memory_constraints ~__context ~vm platformdata =
    same VM twice during memory calculations to determine whether a given VM can start on a particular host..
 *)
 
+let update_platform_secureboot ~__context ~self platform =
+  if List.assoc "secureboot" platform = "auto" then
+    let t = ("secureboot", string_of_bool (Db.Pool.get_uefi_certificates ~__context ~self <> "")) in
+    List.append [t] (List.remove_assoc "secureboot" platform)
+  else
+    platform
+
+let save_uefi_certificates_to_dir ~__context ~self =
+  let dir = try Sys.readdir !Xapi_globs.varstore_dir
+    with Sys_error _ ->
+      Xapi_stdext_unix.Unixext.mkdir_rec "/usr/share/varstored" 0o755;
```

Comment:
This could still fail if `/usr/share/varstored` is not a directory but an existing file. It might be worth checking the exception more carefully and to fail in such a case.